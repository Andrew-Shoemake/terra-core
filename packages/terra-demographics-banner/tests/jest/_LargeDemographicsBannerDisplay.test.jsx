import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { mountWithIntl, shallowWithIntl } from 'terra-enzyme-intl';
import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';
import LargeDemographicsBannerDisplay from '../../src/_LargeDemographicsBannerDisplay';

it('renders large banner that contains all valid information', () => {
  const banner = shallowWithIntl(<LargeDemographicsBannerDisplay
    applicationContent={<span className="risk-score">5%</span>}
    age="25 Years"
    dateOfBirth={{ value: 'May 9, 1993', ariaLabel: 'May 9, 1993' }}
    administrativeSex={{ value: 'Male', ariaLabel: 'Male', isBirthSexMismatch: true }}
    administrativeSexFullText="Administrative Sex"
    identifiers={[
      {
        label: 'MRN',
        value: '12343',
        ariaLabel: 'MRN 1 2 3 4 5',
      },
      {
        label: 'REA',
        value: '3JSDA',
        ariaLabel: 'R E A 3JSDA',
      },
    ]}
    photo=""
    personName="Johnathon Doe"
    preferredFirstName="John"
    dateOfBirthLabel="DOB"
    deceasedDateLabel="Deceased"
    gestationalAgeLabel="GA"
    postMenstrualAgeLabel="PMA"
    sexMismatchFullText="Sex Mismatch"
  />);

  expect(banner).toMatchSnapshot();
});

it('correctly applies the theme context className', () => {
  const banner = mountWithIntl(
    <ThemeContextProvider theme={{ className: 'orion-fusion-theme' }}>
      <LargeDemographicsBannerDisplay />
    </ThemeContextProvider>,
  );
  expect(banner).toMatchSnapshot();
});
