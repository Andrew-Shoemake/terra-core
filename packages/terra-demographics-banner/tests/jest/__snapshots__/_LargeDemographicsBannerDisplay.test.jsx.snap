// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly applies the theme context className 1`] = `
<ThemeContextProvider
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  theme={
    Object {
      "className": "orion-fusion-theme",
    }
  }
>
  <InjectIntl(LargeDemographicsBannerDisplay)>
    <LargeDemographicsBannerDisplay
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
    >
      <section
        className="demographics-banner large-demographics-banner orion-fusion-theme"
        tabIndex={-1}
      >
        <div
          className="content"
        >
          <div
            className="column"
          >
            <h1
              className="person-name"
            />
            <div
              className="person-details"
            >
              <DemographicsBannerValue
                key="age"
              >
                <span
                  className="value"
                >
                  <span
                    aria-hidden={false}
                    className="value-text"
                  />
                </span>
              </DemographicsBannerValue>
              <DemographicsBannerValue
                key="gender"
                valueAriaHidden={false}
              >
                <span
                  className="value"
                >
                  <span
                    aria-hidden={false}
                    className="value-text"
                  />
                </span>
              </DemographicsBannerValue>
              <DemographicsBannerValue
                key="dob"
              >
                <span
                  className="value"
                >
                  <span
                    aria-hidden={false}
                    className="value-text"
                  />
                </span>
              </DemographicsBannerValue>
            </div>
          </div>
          <div
            className="column"
          >
            <div
              className="application-content"
            />
            <div
              className="identifiers"
            />
          </div>
        </div>
      </section>
    </LargeDemographicsBannerDisplay>
  </InjectIntl(LargeDemographicsBannerDisplay)>
</ThemeContextProvider>
`;

exports[`renders large banner that contains all valid information 1`] = `
<LargeDemographicsBannerDisplay
  age="25 Years"
  applicationContent={
    <span
      className="risk-score"
    >
      5%
    </span>
  }
  avatar={
    Object {
      "alt": "Johnathon Doe",
      "color": "neutral",
      "hashValue": "alternative hash",
      "image": "",
      "initials": "JD",
      "isAriaHidden": true,
    }
  }
  dateOfBirth="May 9, 1993"
  dateOfBirthLabel="DOB"
  deceasedDateLabel="Deceased"
  gender="Male"
  gestationalAgeLabel="GA"
  identifiers={
    Array [
      Object {
        "identifierAria": "MRN 1 2 3 4 5",
        "label": "MRN",
        "value": "12345",
      },
      Object {
        "identifierAria": "R E A 3JSDA",
        "label": "REA",
        "value": "3JSDA",
      },
    ]
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  personName="Johnathon Doe"
  postMenstrualAgeLabel="PMA"
  preferredFirstName="John"
/>
`;
